src/main.c

```diff
@@ -44,7 +44,20 @@ void netdata_cleanup_and_exit(int ret)
 	netdata_exit = 1;
 	rrdset_save_all();
 	// kill_childs();
-	unlink(RUN_DIR "/netdata.pid");
+
+	if(pidfile[0]) {
+		if(unlink(pidfile) != 0)
+			error("Cannot unlink pidfile '%s'.", pidfile);
+	}
+
+	if(pidfd != -1) {
```

Comment:
first truncate, close then unlink?

but I still do not understand what is the value to keep the pidfile opened all the time?